### DISCUSSION TITLE
Ricordare ad un utente di spegnere un dispositivo acceso tramite APP

### DISCUSSION THREAD

**Discussion by paolo987** [indomus/forum#1665] [GENERALE - Discussioni generiche]
Buongiorno a tutti,
in azienda ho messo su homeassistant e alcuni relè Shelly per accendere/spegnere dei motori.
Ho installato la APP di homeassistant sullo smartphone dei responsabili di produzione e ora possono accendere e spegnere questi motori anche se sono fuori azienda e fuori dall'orario lavorativo.

Capita che si dimentichino di spegnerli e vorrei fare qualcosa per ricordargli di spegnere il dispositivo.

A me piacerebbe fare in modo che quando un utente dalla APP attiva il dispositivo gli venga chiesto in quale data e ora il dispositivo deve essere spento. Nella data e ora indicate dall'utente un task spegnerà poi il dispositivo.

Qualcuno sa dirmi se questo è fattibile e ha qualche documentazione da darmi per realizzare questo o qualcosa di simile che possa ricordare all'utente che ha acceso un dispositivo che deve anche spegnerlo "prima o poi".

Grazie!

**Comment by indomus** [comment]
Domanda cretina: perché non scrivi [un'automazione](https://indomus.it/formazione/esercizi-di-domotica-su-home-assistant-2-definire-diversi-trigger-per-innescare-automazioni/) che per l'appunto provveda in modo automatico a spegnere quand'è il momento?

**Reply by paolo987** [reply]
Perchè non so quant'è il momento.
Solo l'operatore, quando lo accende, sa che servirà accesso fino all'ora X... è in quel momento che vorrei chiedere all'utente di indicare data e ora di spegnimento.

**Reply by indomus** [reply]
Allora crea un'entità `input_number` o `input_datetime` che l'utente valorizzi da interfaccia (così da usarla come trigger per l'auto spegnimento). Inoltre, un'automazione che invii una notifica quando l'entità da spegnere cambia da spento e acceso.

Oppure più facilmente crea un'automazione arbitraria che a una data ora (chessò, mezzanotte) spenga tutto. Oppure una che dopo un tot di ore dopo l'accensione, faccia spegnimento automatico.
Gli approcci possono essere molteplici.

**Reply by paolo987** [reply]
Mi hai dato una buona idea.
Ci lavoro un po' su e faccio qualche prova.
Grazie

**Reply by paolo987** [reply]
Ho fatto così, ma non mi piace molto.

Un'automazione che quando viene acceso uno switch fa partire un timer.
Una seconda automazione che all'evento timer.finished va a spegnere lo switch.
Ho fatto una plancia per l'utente nel quale vede il proprio switch che può accendere e spegnere.
Ho messo anche il timer nella plancia dell'utente così può vedere fra quanto si spegnerà lo switch... vorrei inoltre che l'utente potesse modificare il timer, impostarlo a suo bisogno ad 1, 2 o n ore ma non è così comodo ed intuitivo.

Cosa potrei usare per rendere più rapido e comodo la modifica del timer?
Grazie per ogni suggerimento.

**Reply by xyzmauro** [reply]
Il timer accetta come duration un integer se lo esprimi in secondi, altrimenti un time percui vedi tu se ti basta usare un input_number settato da 0 a 60 oppure devi utilizzare un input_datetime

https://www.home-assistant.io/integrations/timer/

**Reply by xyzmauro** [reply]
ho appena postato [qui](https://github.com/indomus/forum/discussions/1673#discussion-4047366) un package con una choose e un timer che fa più o meno quel che vuoi tu, prova a prendere spunto

**Reply by paolo987** [reply]
Grazie @xyzmauro il tuo spunto è stato molto utile.
Ho trovato una soluzione che mi soddisfa e dovrebbe essere semplice ed intuitiva per gli utenti

![image](https://user-images.githubusercontent.com/78037015/167104808-27ab47c9-c28d-4051-8733-37857ba535aa.png)

Quando faccio lo start del timer imposto la duration prendendo il valore intero dall'input_number e moltiplicandolo per ottenere il valore in ore sul timer

service: timer.start
data:
  duration: '{{ states(''input_number.fm_compressori_spegnimento'')|int * 60 * 60 }}'
target:
  entity_id: timer.fm_compressori_spegnimento
